.github/workflows/sealights-mpc-test-ci.yaml

```diff
@@ -0,0 +1,66 @@
+name: Validate PR - Testing Phase With SeaLights Monitoring - golang CI
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - main
+jobs:
+  checkout_and_test:
+    name: Golang Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code of the forked repository's head branch
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}  
+          ref: ${{ github.event.pull_request.head.ref }}  
+      - name: Install Go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5
+        with:
+          go-version-file: "./go.mod"
+      - name: Download SeaLights Go agent and CLI tool
+        run: |
+          echo "[Sealights] Downloading Sealights Golang & CLI Agents..."
+          case $(lscpu | awk '/Architecture:/{print $2}') in
+            x86_64) SL_ARCH="linux-amd64";;
+            arm) SL_ARCH="linux-arm64";;
+          esac
+          wget -nv -O sealights-go-agent.tar.gz https://agents.sealights.co/slgoagent/latest/slgoagent-$SL_ARCH.tar.gz
+          wget -nv -O sealights-slcli.tar.gz https://agents.sealights.co/slcli/latest/slcli-$SL_ARCH.tar.gz
+          tar -xzf ./sealights-go-agent.tar.gz && tar -xzf ./sealights-slcli.tar.gz
+          rm -f ./sealights-go-agent.tar.gz ./sealights-slcli.tar.gz
+          ./slgoagent -v 2> /dev/null | grep version && ./slcli -v 2> /dev/null | grep version
+      - name: Write SeaLights token into file
+        run: echo "${SEALIGHTS_AGENT_TOKEN}" > sltoken.txt
+        env:
+          SEALIGHTS_AGENT_TOKEN: '${{secrets.SEALIGHTS_AGENT_TOKEN}}'
+      - name: Fetch branch name for SeaLights commands
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_ENV
+      - name: Create build name for SeaLights commands
+        run: echo "date=$(date +'%y%m%d.%H:%M')" >> $GITHUB_ENV
+      - name: Initiating and configuring SeaLights
+        run: |
+          echo "[Sealights] Initiating and configuring SeaLights to scan the pull request branch"
+          ./slcli config init --lang go --token ./sltoken.txt
+          ./slcli config create-pr-bsid --app multi-platform-controller --target-branch "main" --pull-request-number ${PULL_REQUEST_NAME} --latest-commit ${LATEST_COMMIT_SHA} --repository-url https://github.com/konflux-ci/multi-platform-controller
+        env:
+          PULL_REQUEST_NAME: ${{ github.event.pull_request.number || github.event.issue.number }}
+          LATEST_COMMIT_SHA: ${{github.event.pull_request.head.sha}}
+      - name: Run the SeaLights scan
+        run: |
+          echo "[Sealights] Running the SeaLights scan"
+          ./slcli scan --bsid buildSessionId.txt  --path-to-scanner ./slgoagent --workspacepath ./ --scm git --scmBaseUrl https://github.com/konflux-ci/multi-platform-controller --scmVersion “0” --scmProvider github
```

Comment:
2024-12-16 19:21:00	error	Diff executed with error: code: 400, message: {"error":{"code":1002,"message":"Illegal argument 'bsid d287eacd-7801-4bc8-b901-5e67c4c5463d is of type PULLREQUEST. Pull Request builds are not supported' in field 'undefined'"}}, txid: APIGW-f72958a7-a7a3-4956-893d-c8d102dacbf9	{"module": "cli"}

Im getting this error in my PR scans

Reply:
The result of what? This change needs to be merged so that the second PR with the changes to go-ci.yaml that will trigger this workflow will be put in effect.

Reply:
The result of the sliscan…

Reply:
@meyrevived if you copy-paste the upstream repo's secrets to your fork, you should be able to safely test the `pull_request_target` there and after testing it, share the links to the logs in the upstream repo's PR, which might be helpful for reviewers in the future.